src/cmap/connection_pool.ts

```diff
@@ -408,7 +413,8 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * Pool reset is handled by incrementing the pool's generation count. Any existing connection of a
    * previous generation will eventually be pruned during subsequent checkouts.
    */
-  clear(serviceId?: ObjectId): void {
+  clear(options: { serviceId?: ObjectId; interruptInUseConnections?: boolean } = {}): void {
+    const { serviceId } = options;
```

Comment:
can we move this to be under the comment in L422 and then modify current L423 to only gate on `this.loadBalanced` and throw a runtime error if `serviceId` is not present (just like we do for `generation == null`)? In a real life scenario, this should not happen, and if it did, it would not be correct to fall through to the non-loadbalanced logic; however, in the test scenarios this should help us filter out the tests that aren't relevant for the loadbalanced mode (i.e., all unified tests that call `clear` since for the unified spec, the `clear` is intended to be exclusively the non-loadbalanced clear)

Reply:
done!

Reply:
we'll want to also skip the (correctly) failing LB tests now since those unit tests aren't applicable to LB clear